uaclient/api/tests/test_u_pro_attach_auto_full_auto_attach_v1.py

```diff
@@ -0,0 +1,125 @@
+
+    @mock.patch(
+        "uaclient.actions.enable_entitlement_by_name",
+        return_value=(True, None),
+    )
+    @mock.patch("uaclient.actions.get_cloud_instance")
+    @mock.patch("uaclient.actions.attach_with_token")
+    @mock.patch(
+        M_API + "attach.auto.full_auto_attach.v1._get_auto_attach_token"
+    )
+    def test_error_invalid_ent_names(
+        self,
+        _auto_attach_token,
+        _attach_with_token,
+        _get_cloud_instance,
+        enable_ent_by_name,
+        FakeConfig,
+    ):
+        cfg = FakeConfig(root_mode=True)
+        opts = {
+            "enable": ["esm-infra", "esm-apps", "cis"],
+            "enable_beta": ["realtime-kernel", "test", "wrong"],
+        }
+        options = FullAutoAttachOptions.from_dict(opts)
+        with pytest.raises(exceptions.EntitlementNotFoundError):
+            _full_auto_attach(options, cfg)
+
+        assert 4 == enable_ent_by_name.call_count
+
+    @mock.patch(
+        "uaclient.actions.enable_entitlement_by_name",
+        return_value=(
+            False,
+            CanEnableFailure(
+                CanEnableFailureReason.ALREADY_ENABLED,
+                NamedMessage("test", "test"),
+            ),
+        ),
+    )
+    @mock.patch("uaclient.actions.get_cloud_instance")
+    @mock.patch("uaclient.actions.attach_with_token")
+    @mock.patch(
+        M_API + "attach.auto.full_auto_attach.v1._get_auto_attach_token"
+    )
+    def test_error_ent_not_enabled(
+        self,
+        _auto_attach_token,
+        _attach_with_token,
+        _get_cloud_instance,
+        enable_ent_by_name,
+        FakeConfig,
+    ):
+        cfg = FakeConfig(root_mode=True)
+        opts = {
+            "enable": ["esm-infra", "esm-apps", "cis"],
+            "enable_beta": ["realtime-kernel"],
+        }
+        options = FullAutoAttachOptions.from_dict(opts)
+        with pytest.raises(exceptions.EntitlementNotEnabledError):
+            _full_auto_attach(options, cfg)
+
+        assert 1 == enable_ent_by_name.call_count
+
+    @mock.patch(
+        "uaclient.actions.enable_entitlement_by_name",
+        return_value=(False, None),
+    )
+    @mock.patch("uaclient.actions.get_cloud_instance")
+    @mock.patch("uaclient.actions.attach_with_token")
+    @mock.patch(
+        M_API + "attach.auto.full_auto_attach.v1._get_auto_attach_token"
+    )
+    def test_error_full_auto_attach_fail(
+        self,
+        _auto_attach_token,
+        _attach_with_token,
+        _get_cloud_instance,
+        enable_ent_by_name,
+        FakeConfig,
+    ):
+        cfg = FakeConfig(root_mode=True)
+        opts = {
+            "enable": ["esm-infra", "esm-apps", "fips"],
+            "enable_beta": ["realtime-kernel"],
+        }
+        options = FullAutoAttachOptions.from_dict(opts)
+        with pytest.raises(exceptions.UserFacingError):
+            _full_auto_attach(options, cfg)
+
+        assert 1 >= enable_ent_by_name.call_count
```

Comment:
lets make this stricter
```suggestion
        assert 1 == enable_ent_by_name.call_count
```